.github/workflows/test.yml

```diff
@@ -0,0 +1,108 @@
+name: Integration Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    
+    env:
+      ANSIBLE_CHATBOT_VERSION: 'test'
+      CONTAINER_RUNTIME: 'docker'
+      SERVER_STARTUP_TIMEOUT: '300'
+      UV_CACHE_DIR: /tmp/.uv-cache
+      HF_HOME: /tmp/.cache/huggingface
+
+    steps:
+      - name: Free up disk space
+        uses: jlumbroso/free-disk-space@main
+        with:
+          tool-cache: false
+          android: true
+          dotnet: true
+          haskell: true
+          large-packages: false
+          docker-images: false
+          swap-storage: true
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
+
+      - name: Cache uv dependencies
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.UV_CACHE_DIR }}
+          key: uv-${{ runner.os }}-${{ hashFiles('uv.lock') }}
+          restore-keys: |
+            uv-${{ runner.os }}-
+
+      - name: Cache HuggingFace models
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.HF_HOME }}
+          key: hf-models-${{ runner.os }}-all-MiniLM-L6-v2
+          restore-keys: |
+            hf-models-${{ runner.os }}-
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Setup test data
+        run: make setup-test
+
+      - name: Generate requirements.txt
+        run: make requirements.txt
+
+      - name: Build container image
+        uses: docker/build-push-action@v6
```

Comment:
We can avoid docker and use https://github.com/redhat-actions/buildah-build instead.